src/sentry/nodestore/dynamodb/backend.py

```diff
@@ -0,0 +1,264 @@
+                        'S': id,
+                    },
+                },
+            })
+        )
+        if rv.status == 400:
+            rv = json_loads(rv.data)
+            raise DynamodbException(rv['__type'], rv['message'])
+
+        if rv.status != 200:
+            raise DynamodbException('<unknown>', rv.data)
+
+        data = json_loads(rv.data)
+
+        try:
+            item = data['Item']
+        except KeyError:
+            # DynamoDB won't actually return a 404 or anything
+            # sensible on a miss, it'll just return an empty
+            # dictionary missing 'Item'
+            return None
+
+        return deserialize(item['Data']['B'])
+
+    def get_multi(self, id_list):
+        # In theory, this entire payload is not allowed to be
+        # larger than 16MB for a single response, and we don't
+        # handle this case well, but this should be very unlikely
+        # since a single key isn't allowed to be more than 400KB
+        if len(id_list) == 1:
+            id = id_list[0]
+            return {id: self.get(id)}
+
+        results = {id: None for id in id_list}
+
+        # Max request size is 100 items
+        for chunk in chunked(id_list, 100):
+            rv = self.client.urlopen(
+                target='DynamoDB_20120810.BatchGetItem',
+                body=json_dumps({
+                    'RequestItems': {
+                        self.table: {
+                            'Keys': [
+                                {'Id': {'S': id}}
+                                for id in chunk
+                            ],
+                        },
+                    },
+                })
+            )
+
+            if rv.status != 200:
+                continue
+
+            data = json_loads(rv.data)
+
+            try:
+                items = data['Responses'][self.table]
+            except KeyError:
+                return None
+
+            for item in items:
+                results[item['Id']['S']] = deserialize(item['Data']['B'])
+
+        return results
+
+    def cleanup(self, cutoff_timestamp):
+        while True:
+            rv = self.client.urlopen(
+                target='DynamoDB_20120810.Scan',
+                body=json_dumps({
+                    'TableName': self.table,
+                    'FilterExpression': 'Created < :val',
+                    'ExpressionAttributeValues': {
+                        ':val': {'N': str(datetime_to_unix(cutoff_timestamp))},
+                    }
+                })
+            )
+            if rv.status != 200:
+                return
+
+            data = json_loads(rv.data)
+            if data['Count'] == 0:
+                return
+
+            # TODO(mattrobenolt): parallelize!
+            # This is going to be super slow with a large dataset
+            for item in data['Items']:
+                self.delete(item['Id']['S'])
+
+    def bootstrap(self, write_capacity_units=5, read_capacity_units=5):
```

Comment:
Not sure if we should include this or not since there's not really a CLI interface to fire this off. Could be instructed through the new `sentry exec` though. Something like `sentry exec -c 'from sentry.apps import nodestore; nodestore.bootstrap()'`.